postroj/winrunner.py

```diff
@@ -1,56 +1,118 @@
 # -*- coding: utf-8 -*-
 # (c) 2022 Andreas Motl <andreas.motl@cicerops.de>
 """
-A convenience wrapper around Windows Docker Machine.
+A convenience wrapper around Windows Docker Machine by Stefan Scherer.
 
 https://github.com/StefanScherer/windows-docker-machine
 """
 import json
-import shlex
+import logging
+import os
 import subprocess
+import tempfile
 from pathlib import Path
 from urllib.parse import urlparse
 
-import click
+import appdirs
+import pkg_resources
 
-from postroj.util import port_is_up
+from postroj.util import port_is_up, cmd, fix_tty
+
+logger = logging.getLogger(__name__)
 
 
 class WinRunner:
 
+    #BOX = "2016-box"
     BOX = "2019-box"
+    #BOX = "2022-box"
+    VCPUS = os.environ.get("RACKER_VM_VCPUS", 6)
+    MEMORY = os.environ.get("RACKER_VM_MEMORY", 6144)
+
+    def __init__(self, image: str):
+        self.image_base = image
+        self.image_real = "racker-runtime/" + self.image_base.replace("/", "-")
 
-    def __init__(self):
-        self.workdir = Path.home() / "postroj"
-        self.workdir.mkdir(exist_ok=True)
+        self.workdir = Path(appdirs.user_state_dir(appname="racker"))
+        self.workdir.mkdir(exist_ok=True, parents=True)
 
         self.wdmdir = self.workdir / "windows-docker-machine"
 
     def setup(self):
+        """
+        Prepare virtual machine and adjust system resources.
+        """
         if not self.wdmdir.exists():
-            click.echo("Installing Windows Docker Machine")
+            logger.info(f"Installing Windows Docker Machine into {self.wdmdir}")
             command = f"""
-                cd {self.workdir}
-                git clone https://github.com/StefanScherer/windows-docker-machine
+                cd '{self.workdir}'
+                git clone https://github.com/cicerops/windows-docker-machine --branch racker
```

Comment:
On this spot, we would like to bring in some adjustments to the upstream infrastructure and switch back to mainline afterwards. The corresponding patch is https://github.com/StefanScherer/windows-docker-machine/pull/86.